src/exla/exla.cc

```diff
@@ -79,103 +76,75 @@ static int load(ErlNifEnv* env, void** priv, ERL_NIF_TERM load_info){
-  unsigned atom_length;
-  // Default to Host
-  if(!enif_get_atom_length(env, term, &atom_length, ERL_NIF_LATIN1)) return xla::PlatformUtil::GetPlatform("Host").ConsumeValueOrDie();
-
-  std::string atom_str;
-  atom_str.resize(atom_length+1);
+int enif_get_platform(ErlNifEnv* env, ERL_NIF_TERM term, stream_executor::Platform* platform){
+  std::string platform_str;
 
-  if(!enif_get_atom(env, term, &(*(atom_str.begin())), atom_str.size(), ERL_NIF_LATIN1)) return xla::PlatformUtil::GetPlatform("Host").ConsumeValueOrDie();
-
-  atom_str.resize(atom_length);
-
-  if(atom_str.compare("host") == 0){
-    return xla::PlatformUtil::GetPlatform("Host").ConsumeValueOrDie();
-  } else if(atom_str.compare("cuda") == 0){
-    return xla::PlatformUtil::GetPlatform("Cuda").ConsumeValueOrDie();
-  } else {
-    return xla::PlatformUtil::GetPlatform("Host").ConsumeValueOrDie();
+  if(!ExlaNifUtil::get_atom(env, term, platform_str)){
+    return 0;
   }
-}
 
-xla::PrimitiveType enif_get_primitive_type(ErlNifEnv* env, ERL_NIF_TERM term){
-  unsigned atom_length;
-  if(!enif_get_atom_length(env, term, &atom_length, ERL_NIF_LATIN1)) return xla::PrimitiveType::PRIMITIVE_TYPE_INVALID;
-
-  std::string atom_str;
-  atom_str.resize(atom_length+1);
-
-  if(!enif_get_atom(env, term, &(*(atom_str.begin())), atom_str.size(), ERL_NIF_LATIN1)) return xla::PrimitiveType::PRIMITIVE_TYPE_INVALID;
-
-  atom_str.resize(atom_length);
-
-  if(atom_str.compare("pred") == 0){
-    return xla::PrimitiveType::PRED;
-  } else if(atom_str.compare("int8") == 0){
-    return xla::PrimitiveType::S8;
-  } else if(atom_str.compare("int16") == 0){
-    return xla::PrimitiveType::S16;
-  } else if(atom_str.compare("int32") == 0){
-    return xla::PrimitiveType::S32;
-  } else if(atom_str.compare("int64") == 0){
-    return xla::PrimitiveType::S64;
-  } else if(atom_str.compare("uint8") == 0){
-    return xla::PrimitiveType::U8;
-  } else if(atom_str.compare("uint16") == 0){
-    return xla::PrimitiveType::U16;
-  } else if(atom_str.compare("uint32") == 0){
-    return xla::PrimitiveType::U32;
-  } else if(atom_str.compare("uint64") == 0){
-    return xla::PrimitiveType::U64;
-  } else if(atom_str.compare("float16") == 0){
-    return xla::PrimitiveType::F16;
-  } else if(atom_str.compare("bfloat16") == 0){
-    return xla::PrimitiveType::BF16;
-  } else if(atom_str.compare("float32") == 0){
-    return xla::PrimitiveType::F32;
-  } else if(atom_str.compare("float64") == 0){
-    return xla::PrimitiveType::F64;
-  } else if(atom_str.compare("complex64") == 0){
-    return xla::PrimitiveType::C64;
-  } else if(atom_str.compare("complex128") == 0){
-    return xla::PrimitiveType::C128;
-  } else if(atom_str.compare("tuple") == 0){
-    return xla::PrimitiveType::TUPLE;
-  } else if(atom_str.compare("opaque") == 0){
-    return xla::PrimitiveType::OPAQUE_TYPE;
-  } else if(atom_str.compare("token") == 0){
-    return xla::PrimitiveType::TOKEN;
+  if(platform_str.compare("host") == 0){
```

Comment:
I wonder if we should consider other approaches here when it comes to our atom handling. Some options are:

1. Convert atoms into strings in Elixir itself to reduce the code here
2. Convert atoms into integers in Elixir itself and match on integers here

In both cases, I would like the validation of the data to belong to Elixir as much as possible, so we get both good error messages and reduce the amount of C++ code. This will be handy when we do the changes we discussed today of having a EXLA.NIF module. What do you think?

For now, you can either:

1. Leave this as is but add a TODO

2. Rename this function to get_int_platform and introduce a Exla.get_platform in Elixir that does the conversion and call Exla.get_int_plataform (until we introduce the Exla.NIF module)